wrapper/src/main/java/software/amazon/jdbc/plugin/AuroraConnectionTrackerPlugin.java

```diff
@@ -94,6 +97,11 @@ public Connection connectInternal(
 
     if (conn != null) {
       if (!rdsHelper.isRdsInstance(currentHostSpec.getHost())) {
+        // Remove pre-existing instance endpoint.
+        currentHostSpec.getAliases().stream()
```

Comment:
Why not to remove all of them?

Reply:
Should be at most 1 instance endpoint here.